src/main/java/com/github/nkzawa/engineio/client/transports/WebSocket.java

```diff
@@ -110,10 +136,14 @@ protected void write(Packet[] packets) {
             Parser.encodePacket(packet, new Parser.EncodeCallback() {
                 @Override
                 public void call(Object packet) {
-                    if (packet instanceof String) {
-                        self.ws.send((String) packet);
-                    } else if (packet instanceof byte[]) {
-                        self.ws.send((byte[]) packet);
+                    try {
+                        if (packet instanceof String) {
+                            self.ws.sendMessage(TEXT, new Buffer().writeUtf8((String) packet));
+                        } else if (packet instanceof byte[]) {
+                            self.ws.sendMessage(BINARY, new Buffer().write((byte[]) packet));
+                        }
```

Comment:
@niqo01 I add you as collaborator, you could modify them directly :)
I think you are more familiar with websocket then I.